config/boot.rb

```diff
@@ -24,9 +25,21 @@
 
 begin
   require 'bundler/setup'
-rescue LoadError
-  $stderr.puts "[*] Metasploit requires the Bundler gem to be installed"
-  $stderr.puts "    $ gem install bundler"
+rescue LoadError => e
+  msg = e.to_s
+  ver = msg.scanf("You have already activated bundler %d.%d.%d, but your Gemfile requires bundler %d.%d.%d.")
+  if ver.length == 6
+    activated_ver = "#{ver[0]}.#{ver[1]}.#{ver[2]}"
+    wanted_ver = "#{ver[3]}.#{ver[4]}.#{ver[5]}"
+    $stderr.puts "[*] Bundler #{activated_ver} conflicts with Ruby #{RUBY_VERSION}'s default of #{wanted_ver}."
```

Comment:
I think tying this to the rubygems version, `Gem::VERSION` (which is independent of Ruby and the source of the conflicts), would be better, since this also crops up where people or systems did `gem update --system` on non-2.6 installs. Also, this seems like a generic way of solving this one problem without providing the additional help we can (a `gem install bundler` would also work) even though we only have the one case to deal with and no indication that future cases will be similar. We can recognize this specific case because `Bundler::VERSION` will be defined and `1.17.2` (even if that version is not installed) and `Gem::VERSION` will start with `3.0`. Having specific code for these versions will also allow us to remove the it once this passes.

Reply:
Your suggestion is fine, I welcome a PR from you implementing it. But all I wanted to do was print the darn exception, since the problem is never actually 'I forgot to install bundler' these days.

Reply:
I'm reverting all of the complexity from this PR, changing it to just print the exception.